tests/unit/commands/local/lib/test_cfn_api_provider.py

```diff
@@ -1187,6 +1187,127 @@ def test_empty_integration_array(self):
         self.assertIsNone(provider.api.cors)
 
 
+class TestExtractResourcesWithDisableAuthorizerFlag(TestCase):
+    @parameterized.expand(
+        [
+            ("when enabled doesn't extract authorizer", True),
+            ("when disabled extracts authorizer", False),
+        ]
+    )
+    @patch("samcli.lib.providers.cfn_api_provider.CfnApiProvider._get_route_function_name")
+    def test_extract_resources_apigw_v2_route(self, _, disable_authorizer, mock_get_route_function_name: Mock):
+        template = {
+            "Resources": {
+                "MyRoute": {
+                    "Type": "AWS::ApiGatewayV2::Route",
+                    "Properties": {
+                        "ApiId": "MockApiId",
+                        "AuthorizationType": "CUSTOM",
+                        "RouteKey": "Get /Test",
+                        "Target": "DummyTarget",
+                        "AuthorizerId": "AuthorizerId",
+                        "OperationName": "MockOperation",
+                    },
+                },
+            }
+        }
+        mock_get_route_function_name.return_value = ("Dummy", "Values")
+        provider = CfnApiProvider()
+        mock_collector = Mock()
+        mock_collector.add_routes = Mock()
+        provider._extract_cfn_gateway_v2_route(
+            stack_path="stack_path",
+            resources={},
+            logical_id="MyRoute",
+            route_resource=template["Resources"]["MyRoute"],
+            collector=mock_collector,
+            disable_authorizer=disable_authorizer,
+        )
+
+        if disable_authorizer:
+            route = mock_collector.add_routes.call_args[0][1][0]
+            assert route.authorizer_name is None
+
+        if not disable_authorizer:
+            mock_collector.add_routes.assert_called()
+            route = mock_collector.add_routes.call_args[0][1][0]
+            assert route.authorizer_name is "AuthorizerId"
```

Comment:
Could we utilize `unittest`'s assertions here instead of using Python's `assert`?

Reply:
hey matey, do you mean self.assertEqual sort of assertions?

Reply:
Yup, more so utilizing the `.assert_called_with(<args>)` method (ref: https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_with).

Reply:
is there a way to partial match a route object, or do I have create a route object to match exactly?

Reply:
`unittest` provides an `ANY` object (https://docs.python.org/3/library/unittest.mock.html#any) that can be used in place of a specific argument check

Reply:
but what about for object properties e.g.  mock_collector.add_routes.assert_called_once_with(ANY, [mock_route(ANY, authorizer_name="AuthorizerId")])

doesn't seem to have the intended effect.

Reply:
Ah I see, sorry for the initial misunderstanding. Getting the object with `call_args` should be fine in this case then, using `assert_called_once_with` would involve mocking out a `Route` object. We can go with your initial suggestion of `self.assertEqual`